ckanext/opendata_theme/processor.py

```diff
@@ -0,0 +1,397 @@
+        return '#131517'
+
+
+class LinkHoverColor(AbstractParser):
+    def get_css_from_data(self, data):
+        self.parse_form_data(data)
+        if self.color:
+            return {'color': self.color.encode('utf-8')}
+
+    def get_class_name(self):
+        return 'a:hover'
+
+    def get_form_name(self):
+        return 'custom-css-link-hover-color'
+
+    def get_title(self):
+        return 'Link Hover Color'
+
+    def get_default_color(self):
+        return '#165cab'
+
+
+class FooterLinkColor(AbstractParser):
+    def get_css_from_data(self, data):
+        self.parse_form_data(data)
+        if self.color:
+            return {'color': self.color.encode('utf-8')}
+
+    def get_class_name(self):
+        return ('.site-footer a,'
+                '.site-footer a:hover')
+
+    def get_form_name(self):
+        return 'custom-css-footer-link-color'
+
+    def get_title(self):
+        return 'Footer Link Color'
+
+    def get_default_color(self):
+        return '#ffffff'
+
+
+class CustomStyleProcessor:
+    def __init__(self):
+        self.processor_account_header_background_color = AccountHeaderBackGroundColor()
+        self.processor_account_header_text_color = AccountHeaderTextColor()
+
+        self.processor_account_hover_background_color = AccountHoverBackgroundColor()
+
+        self.processor_navigation_header_background_color = NavigationHeaderBackGroundColor()
+        self.processor_navigation_header_text_color = NavigationHeaderTextColor()
+
+        self.processor_navigation_hover_background_color = NavigationButtonHoverBackgroundColor()
+
+        self.processor_module_header_background_color = ModuleHeaderBackgroundColor()
+        self.processor_module_header_text_color = ModuleHeaderTextColor()
+
+        self.processor_footer_background_color = FooterBackGroundColor()
+        self.processor_footer_text_color = FooterTextColor()
+
+        self.processor_link_color = LinkColor()
+        self.processor_link_hover_color = LinkHoverColor()
+
+        self.processor_footer_link_color = FooterLinkColor()
+
+        self.processors = [
+            self.processor_account_header_background_color,
+            self.processor_account_header_text_color,
+
+            self.processor_account_hover_background_color,
+
+            self.processor_navigation_header_background_color,
+            self.processor_navigation_header_text_color,
+
+            self.processor_navigation_hover_background_color,
+
+            self.processor_module_header_background_color,
+            self.processor_module_header_text_color,
+
+            self.processor_footer_background_color,
+            self.processor_footer_text_color,
+
+            self.processor_link_color,
+            self.processor_link_hover_color,
+
+            self.processor_footer_link_color
+        ]
+        self.add_position_to_processors()
+
+
+    def add_position_to_processors(self):
+        for i, processor in enumerate(self.processors):
```

Comment:
lists and tuples are keeping the order of objects. it looks like an unnecessary action.